docs/source/io/introduction.rst

```diff
@@ -0,0 +1,27 @@
+io
+===
+
+The io package provides a traited |File| object provides properties and methods
+for common file path manipulation operations.  On Python 3 much of this
+functionality is provided by the `pathlib`_ module, and for new code we
+encourage users to investigate if `pathlib`_ can satisfy their usecase before
+they turn to the `apptools.io` |File| object
+
+io.h5
+-----
+The io.h5 sub-package provides a wrapper around `PyTables`_ with a
+dictionary-style mapping.  Note that this module has a decent amount of
+overlap with `zarr`_.
```

Comment:
We probably want to add more specifics here, but I am not exactly confident in what `io.h5` uniquely provides...